File: Data Structures & Algorithms/find-duplicate-integer/submission-0.cpp
class Solution {
public:
    int findDuplicate(vector<int>& nums) {
        int a = 0, b = nums[0] ;

        while (true) {
            if (a == nums[b]) {
                return a ;
            }

            a = b ;
            b = nums[a] ;
        }

    }
};

File: Data Structures & Algorithms/house-robber/submission-0.py
class Solution:
    def rob(self, nums: List[int]) -> int:
        def solve(index) : 
            if index <= -1 :
                return 0
            

            return max(nums[index] + solve(index - 2), solve(index - 1))

        
        return solve(len(nums) - 1)